src/AudioOutputI2S.cpp

```diff
@@ -298,9 +298,9 @@ bool AudioOutputI2S::ConsumeSample(int16_t sample[2])
     uint32_t s32;
     if (output_mode == INTERNAL_DAC)
     {
-      int16_t l = Amplify(ms[LEFTCHANNEL]) + 0x8000;
-      int16_t r = Amplify(ms[RIGHTCHANNEL]) + 0x8000;
-      s32 = (r << 16) | (l & 0xffff);
+      uint32_t l = ((int32_t)Amplify(ms[LEFTCHANNEL])) + 0x8000;
```

Comment:
Could we do saturation like in the MOD?  OTW I'm not sure that doing the math as 32bit really changes anything here since we're a) adding the 0x8000 DC offset and b) we're just truncating the lower bits anyway, same as `int16_t` math would do...

Reply:
I'll think about it ...
The point is that we want to go from "AC" to "DC", by adding -INT16_MIN to the signed value. Technicially he result will not fit into int16, however an uint16 can hold it. That's why I think that doing it in int32 is a safer solution. On the other side, maybe the code from the MOD mixer could also work here.

In general, is my understanding correct that the audio output class should receive samples as signed int16 ?

Reply:
It actually is receiving signed, fractional 16-bit samples, with the binary point between b15/b14 (i.e. 0/1.xxxxx).  Basically what the I2S DACs expect.

Reply:
Let me check tomorrow. Actually this modification is not so important, and we could  (possibly) leave the fuction as-is. 
I just modified this to be 100% sure that the I2S output is not the cause of overflow noise.

Reply:
just checked - the original code from AudioOutputI2S.cpp works, too. At least I did not hear noisy artefacts even with audio input that uses the full range.

If you prefer, I can remove this modification from my PR.

Reply:
Could you remove it, please?   Thx!

Reply:
OK, done.